frontend/src/components/ViewTrips/trip-utils.test.js

```diff
@@ -107,3 +96,79 @@ describe('moveCurUserEmailToFront tests', () => {
     expect(testEmailOutputArr).toEqual(expectedEmailArr);
   });
 });
+
+// Note: Although getCurAcceptedCollabArr handles array's of uids, the strings
+//       in each array do not have to have a specific form. Thus, the
+//       user emails can be used as test string array` elements.
+describe('getCurAcceptedCollabArr tests', () => {
+  test('No previous accepted collabs', () => {
+    const expectedNewAccepCollabUidArr = [];
+    const testFormCollabUidArr = [USER_A_EMAIL, USER_B_EMAIL, USER_Z_EMAIL];
+    const testCurAccepCollabUidArr = [];
+
+    const testNewAccepCollabUidArr = TripUtils.getCurAcceptedCollabUidArr(
+                                testFormCollabUidArr, testCurAccepCollabUidArr);
+
+    expect(testNewAccepCollabUidArr).toEqual(expectedNewAccepCollabUidArr);
+  });
+
+  test('One previous accepted collab', () => {
+    const expectedNewAccepCollabUidArr = [USER_A_EMAIL];
+    const testFormCollabUidArr = [USER_A_EMAIL, USER_B_EMAIL, USER_Z_EMAIL];
+    const testCurAccepCollabUidArr = [USER_A_EMAIL];
+
+    const testNewAccepCollabUidArr = TripUtils.getCurAcceptedCollabUidArr(
+                                testFormCollabUidArr, testCurAccepCollabUidArr);
+
+    expect(testNewAccepCollabUidArr).toEqual(expectedNewAccepCollabUidArr);
+  });
+
+  test('Two previous accepted collab and one deleted in current collabs', () => {
+    const expectedNewAccepCollabUidArr = [USER_B_EMAIL];
+    const testFormCollabUidArr = [USER_B_EMAIL, USER_Z_EMAIL];
+    const testCurAccepCollabUidArr = [USER_A_EMAIL, USER_B_EMAIL];
+
+    const testNewAccepCollabUidArr = TripUtils.getCurAcceptedCollabUidArr(
+                                testFormCollabUidArr, testCurAccepCollabUidArr);
+
+    expect(testNewAccepCollabUidArr).toEqual(expectedNewAccepCollabUidArr);
+  });
+});
+
+// Note: Although getCurPendingCollabArr handles array's of uids, the strings
+//       in each array do not have to have a specific form. Thus, the
+//       user emails can be used as test string array` elements.
+describe('getCurPending CollabArr tests', () => {
+  test('No previous accepted collabs', () => {
+    const expectedNewAccepCollabUidArr = [USER_A_EMAIL, USER_B_EMAIL, USER_Z_EMAIL];
+    const testFormCollabUidArr = [USER_A_EMAIL, USER_B_EMAIL, USER_Z_EMAIL];
+    const testCurAccepCollabUidArr = [];
+
+    const testNewAccepCollabUidArr = TripUtils.getCurPendingCollabUidArr(
+                                testFormCollabUidArr, testCurAccepCollabUidArr);
+
+    expect(testNewAccepCollabUidArr).toEqual(expectedNewAccepCollabUidArr);
+  });
+
+  test('One previous accepted collab', () => {
```

Comment:
Are any pending?

Reply:
I made a note above the function about this because I could see how this may be confusing. In short, both `getNewPendingCollabUidArr` and `getNewAcceptedCollabUidArr` use the current accepted collabs to determine the next state. Thus it is only important to mention what the current accepted collaborators are for the test name.